pkg/dbg/dbg.go

```diff
@@ -0,0 +1,96 @@
+package dbg
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"runtime"
+	"runtime/pprof"
+	"syscall"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"path"
+)
+
+const (
+	fnameFmt       = "2006-01-02T15:04:05.999999-0700MST"
+	stackTraceSize = 5120 * 1024 // 5MB
+)
+
+// Initialize the debug handlers
+func Init(name string, dir string) {
+	installSignalHandlers(name, dir)
+}
+
+func installSignalHandlers(name string, dir string) {
+	dumpChannel := make(chan os.Signal, 1)
+	cpuProfChannel := make(chan os.Signal, 1)
+	signal.Notify(dumpChannel, syscall.SIGUSR1)
+	go func() {
+		for {
+			<-dumpChannel
+			dumpGoMemoryTrace()
+			dumpHeap(generateFilename(name, dir, ".heap"))
+			dumpGoProfile(generateFilename(name, dir, ".stack"))
+		}
+	}()
+	signal.Notify(cpuProfChannel, syscall.SIGUSR2)
+	go func() {
+		started := false
+		for {
+			<-cpuProfChannel
+			if !started {
+				startCPUProfileDump(generateFilename(name, dir, ".cpuprof"))
+				started = true
+			} else {
+				pprof.StopCPUProfile()
+				started = false
+			}
+		}
+	}()
+
+}
+
+func generateFilename(name string, dir string, suffix string) string {
+	return path.Join(dir, name+"."+time.Now().Format(fnameFmt)+suffix)
+}
+
+func startCPUProfileDump(filename string) {
+	f, err := os.Create(filename)
+	if err != nil {
+		logrus.Errorf("could not create memory profile: %v", err)
```

Comment:
this should be "could not create CPU profile"

Reply:
Fixed